src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1783,20 +1784,21 @@ private static IsNullOperation MakeIsNullOperation(SemanticModel semanticModel,
                                        constantValue.HasValue ? new Optional<object>(constantValue.Value == null) : default);
         }
 
-        private static IOperation TryUnwrapNullableValue(IOperation value)
+        private static IOperation TryUnwrapNullableValue(IOperation value, SemanticModel semanticModel)
         {
             ITypeSymbol valueType = value.Type;
 
             Debug.Assert(ITypeSymbolHelpers.IsNullableType(valueType));
 
-            foreach (ISymbol candidate in valueType.GetMembers("GetValueOrDefault"))
+            var method = (IMethodSymbol)semanticModel.Compilation.CommonGetSpecialTypeMember(SpecialMember.System_Nullable_T_GetValueOrDefault);
+
+            if (method != null)
             {
-                if (candidate.Kind == SymbolKind.Method && !candidate.IsStatic && candidate.DeclaredAccessibility == Accessibility.Public)
+                foreach (ISymbol candidate in valueType.GetMembers(method.Name))
                 {
-                    var method = (IMethodSymbol)candidate;
-                    if (method.Parameters.Length == 0 && method.RefKind == RefKind.None &&
-                        method.OriginalDefinition.ReturnType.Equals(((INamedTypeSymbol)valueType).OriginalDefinition.TypeParameters[0]))
+                    if (candidate.OriginalDefinition.Equals(method))
                     {
+                        method = (IMethodSymbol)candidate;
                         return new InvocationExpression(method, value, isVirtual: false,
```

Comment:
It might be useful to extract this logic into a helper extension method `ISymbol.AsMember(ITypeSymbol)`

Reply:
> It might be useful to extract this logic into a helper extension method ISymbol.AsMember(ITypeSymbol)

At the moment the two use sites actually iterate over different sets of members. I'd rather keep them as is, but I will keep an eye of usage in the future. 

---
In reply to: [179538009](https://github.com/dotnet/roslyn/pull/25916#discussion_r179538009) [](ancestors = 179538009)